integration/ingester_test.go

```diff
@@ -0,0 +1,336 @@
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/integration/ingester_sharding_test.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+//go:build requires_docker
+
+package integration
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/util/test"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+func TestIngesterMixedFloatHistogramSeries(t *testing.T) {
+	query := "foobar"
+	queryEnd := time.Now().Round(time.Second)
+	queryStart := queryEnd.Add(-1 * time.Hour)
+	queryStep := 10 * time.Minute
+
+	testCases := map[string]struct {
+		inSeries []prompb.TimeSeries
+		expected model.Matrix
+	}{
+		"float series": {
+			inSeries: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{
+							Name:  "__name__",
+							Value: query,
+						},
+					},
+					Samples: []prompb.Sample{
+						{
+							Timestamp: queryStart.UnixMilli(),
+							Value:     100,
+						},
+						{
+							Timestamp: queryStart.Add(queryStep).UnixMilli(),
+							Value:     110,
+						},
+						{
+							Timestamp: queryStart.Add(queryStep * 2).UnixMilli(),
+							Value:     120,
+						},
+						{
+							Timestamp: queryStart.Add(queryStep * 3).UnixMilli(),
+							Value:     130,
+						},
+						{
+							Timestamp: queryStart.Add(queryStep * 4).UnixMilli(),
+							Value:     140,
+						},
+						{
+							Timestamp: queryStart.Add(queryStep * 5).UnixMilli(),
+							Value:     150,
+						},
+					},
+				},
+			},
+			expected: model.Matrix{
+				{
+					Metric: model.Metric(map[model.LabelName]model.LabelValue{"__name__": "foobar"}),
+					Values: []model.SamplePair{
+						{
+							Timestamp: model.Time(queryStart.UnixMilli()),
+							Value:     model.SampleValue(100),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep).UnixMilli()),
+							Value:     model.SampleValue(110),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 2).UnixMilli()),
+							Value:     model.SampleValue(120),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 3).UnixMilli()),
+							Value:     model.SampleValue(130),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 4).UnixMilli()),
+							Value:     model.SampleValue(140),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 5).UnixMilli()),
+							Value:     model.SampleValue(150),
+						},
+					},
+				},
+			},
+		},
+		"native histogram series": {
+			inSeries: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{
+							Name:  "__name__",
+							Value: query,
```

Comment:
While the variable is the same, it is confusing to have `query` here (given the `expected` doesn't use `query` too)
```suggestion
							Value: "foobar",
```